native/cocos/bindings/event/EventDispatcher.cpp

```diff
@@ -289,49 +293,52 @@ void EventDispatcher::dispatchOrientationChangeEvent(int orientation) {
 }
 
 void EventDispatcher::dispatchEnterBackgroundEvent() {
-    EventDispatcher::doDispatchEvent(EVENT_COME_TO_BACKGROUND, "onPause", se::EmptyValueArray);
+    EventDispatcher::doDispatchJsEvent("onPause", se::EmptyValueArray);
+    EventDispatcher::dispatchCustomEvent(EVENT_COME_TO_BACKGROUND, 0);
 }
 
 void EventDispatcher::dispatchEnterForegroundEvent() {
-    EventDispatcher::doDispatchEvent(EVENT_COME_TO_FOREGROUND, "onResume", se::EmptyValueArray);
+    EventDispatcher::doDispatchJsEvent("onResume", se::EmptyValueArray);
+    EventDispatcher::dispatchCustomEvent(EVENT_COME_TO_FOREGROUND, 0);
 }
 
 void EventDispatcher::dispatchMemoryWarningEvent() {
-    EventDispatcher::doDispatchEvent(EVENT_MEMORY_WARNING, "onMemoryWarning", se::EmptyValueArray);
+    EventDispatcher::doDispatchJsEvent("onMemoryWarning", se::EmptyValueArray);
+    EventDispatcher::dispatchCustomEvent(EVENT_MEMORY_WARNING, 0);
 }
 
 void EventDispatcher::dispatchRestartVM() {
-    EventDispatcher::doDispatchEvent(EVENT_RESTART_VM, "onRestartVM", se::EmptyValueArray);
+    EventDispatcher::doDispatchJsEvent("onRestartVM", se::EmptyValueArray);
+    EventDispatcher::dispatchCustomEvent(EVENT_RESTART_VM, 0);
 }
 
 void EventDispatcher::dispatchCloseEvent() {
-    EventDispatcher::doDispatchEvent(EVENT_CLOSE, "onClose", se::EmptyValueArray);
+    EventDispatcher::doDispatchJsEvent("onClose", se::EmptyValueArray);
+    EventDispatcher::dispatchCustomEvent(EVENT_CLOSE, 0);
 }
 
 void EventDispatcher::dispatchDestroyWindowEvent() {
-    EventDispatcher::doDispatchEvent(EVENT_DESTROY_WINDOW, "", se::EmptyValueArray);
+    EventDispatcher::doDispatchJsEvent("", se::EmptyValueArray);
```

Comment:
"" -> nullptr

Reply:
As the `nullptr` means no jsFunction will be called, just delete it.